lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -153,43 +153,78 @@ class TestAuthorizationView(TestCase):
     def setUp(self):
         super(TestAuthorizationView, self).setUp()
         self.user = UserFactory()
-        self.dop_client = self._create_confidential_client(user=self.user, client_id='dop-app-client-id')
-
-    def _create_confidential_client(self, user, client_id):
-        """
-        Create a confidential client suitable for testing purposes.
-        """
-        return self.dop_adapter.create_confidential_client(
-            name='test_app',
-            user=user,
-            client_id=client_id,
-            redirect_uri=DUMMY_REDIRECT_URL
+        self.dot_app = self.dot_adapter.create_confidential_client(
+            name='test dot application',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='confidential-dot-app-client-id',
+        )
+        self.dop_app = self.dop_adapter.create_confidential_client(
+            name='test dop client',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='confidential-dop-app-client-id',
         )
 
-    def test_authorization_view(self):
+    @ddt.data('dop', 'dot')
+    def test_post_authorization_view(self, client_type):
+        oauth_application = getattr(self, '{}_app'.format(client_type))
         self.client.login(username=self.user.username, password='test')
+        allow_field = {'dop': 'authorize', 'dot': 'allow'}
         response = self.client.post(
             '/oauth2/authorize/',
             {
-                'client_id': self.dop_client.client_id,  # TODO: DOT is not yet supported (MA-2124)
+                'client_id': oauth_application.client_id,
                 'response_type': 'code',
                 'state': 'random_state_string',
                 'redirect_uri': DUMMY_REDIRECT_URL,
+                'scope': 'profile email',
+                allow_field[client_type]: True,
             },
             follow=True,
         )
 
+        check_response = getattr(self, '_check_{}_response'.format(client_type))
+        check_response(response)
+
+    def _check_dot_response(self, response):
+        """
+        Check that django-oauth-toolkit gives an appropriate authorization response.
+        """
+        # django-oauth-toolkit tries to redirect to the user's redirect URL
+        self.assertEqual(response.status_code, 404)  # We used a non-existent redirect url.
+        expected_redirect_prefix = u'{}?'.format(DUMMY_REDIRECT_URL)
+        self._assert_startswith(self._redirect_destination(response), expected_redirect_prefix)
```

Comment:
In this case, I can't use assertRedirects, because the redirect URL includes the grant code that was randomly generated by this request, so I'm just checking the prefix of the target URL instead.  I will update the DOP one to use it though, because that's a fully known URL.